modules/engine/src/scenegraph/group-node.ts

```diff
@@ -108,4 +108,20 @@ export class GroupNode extends ScenegraphNode {
       }
     }
   }
+
+  preorderTraversal(
```

Comment:
What is a `preorderTraversal`? Should we have an option parameter to traverse instead?

```ts
traverse(... {..., traversalOrder: 'breadth-first' | 'depth-first'})
```

or something like that?